src/main/java/com/liferay/faces/util/cache/CacheFactory.java

```diff
@@ -0,0 +1,228 @@
+/**
+ * Copyright (c) 2000-2017 Liferay, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.liferay.faces.util.cache;
+
+import javax.faces.context.ExternalContext;
+
+import com.liferay.faces.util.factory.FactoryExtensionFinder;
+import com.liferay.faces.util.helper.Wrapper;
+
+
+/**
+ * @author  Kyle Stiemann
+ * @since   3.1
+ * @since   2.1
+ * @since   1.1
+ */
+public abstract class CacheFactory implements Wrapper<CacheFactory> {
+
+	/**
+	 * Returns a new instance of {@link Cache} from the {@link CacheFactory} found by the {@link
+	 * FactoryExtensionFinder}. The returned instance is designed to be used during execution of a request thread, so it
+	 * is not guaranteed to be {@link java.io.Serializable}.
```

Comment:
@stiemannkj1 In other JavaDoc we have stated whether or not the returned instance is thread-safe, and whether or not it is guaranteed to be serializable. But we did not link/tie those two concepts together like we have here using the phrase "so it is" -- recommend that we follow the pattern consistently. For example, see https://github.com/liferay/liferay-faces-bridge-api/blob/master/bridge-api/src/main/java/javax/portlet/faces/BridgeEventHandlerFactory.java#L28